x-pack/elastic-agent/pkg/composable/providers/agent/agent.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package agent
+
+import (
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/application/info"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/composable"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
+)
+
+func init() {
+	composable.Providers.AddContextProvider("agent", ContextProviderBuilder)
+}
+
+type contextProvider struct{}
+
+// Run runs the Agent context provider.
+func (*contextProvider) Run(comm composable.ContextProviderComm) error {
+	a, err := info.NewAgentInfo()
+	if err != nil {
+		return err
+	}
+	comm.Set(map[string]interface{}{
+		"id": a.AgentID(),
+		"version": map[string]interface{}{
+			"version":    release.Version(),
+			"commit":     release.Commit(),
+			"build_time": release.BuildTime().Format("2006-01-02 15:04:05 -0700 MST"),
+			"snapshot":   release.Snapshot(),
+		},
+	})
+	return nil
+}
+
+// ContextProviderBuilder builds the context provider.
+func ContextProviderBuilder(_ *config.Config) (composable.ContextProvider, error) {
+	return &contextProvider{}, nil
```

Comment:
Here we go with passing the config. So the context provider must support config.Config to be passed in?

Reply:
Both dynamic or context provider can take a config. Its up to the provider to parse it and use it, if they want.

Both `local` (context) and `local_dynamic` (dynamic) read the config to get the vars.